core/src/main/java/de/bwaldvogel/mongo/backend/QueryResult.java

```diff
@@ -0,0 +1,43 @@
+package de.bwaldvogel.mongo.backend;
+
+import java.util.Collections;
+import java.util.Iterator;
+
+import de.bwaldvogel.mongo.bson.Document;
+
+public class QueryResult<T extends Document> implements Iterable<T> {
+
+    private Iterable<T> documents;
+    private int remainingDocuments;
+    private long cursorId;
+
+    public QueryResult() {
+        this(Collections.emptyList(), 0);
+    }
+
+    public QueryResult(Iterable<T> documents, long cursorId) {
+        this.documents = documents;
+        this.cursorId = cursorId;
+    }
+
+    public QueryResult(Iterable<T> documents) {
+        this(documents, 0);
+    }
+
+    public Iterable<T> getDocuments() {
+        return documents;
+    }
+
+    public int getRemainingDocuments() {
+        return remainingDocuments;
+    }
+
+    @Override
```

Comment:
I created QueryResult because besides the Documents resulting from the queries I needed the cursorId to include it in the response sent over the wire. 
The cursor will hold the remining documents that were not returned, so the subsequent iterations (GET_MORE) will get the result from the cursor.
The cursorId in the Cursor class is a unique identifier of the cursor within the collection (actually unique in general)